tests/test_lexer.py

```diff
@@ -0,0 +1,277 @@
+
+        registry: TokenRegistry = TokenRegistry()
+        registry.register(lambda x: AToken(x), r'a')
+        registry.register(lambda x: AAToken(x), r'aa')
+
+        match = registry.get_token('aaa')
+        assert match is not None
+        assert isinstance(match[0]('aa'), AAToken)
+
+    def test_get_token_multiple_inverted(self) -> None:
+        class AToken(Token):
+            pass
+
+        class AAToken(Token):
+            pass
+
+        registry: TokenRegistry = TokenRegistry()
+        registry.register(lambda x: AToken(x), r'a')
+        registry.register(lambda x: AAToken(x), r'aa')
+
+        match = registry.get_token('aaa')
+        assert match is not None
+        assert isinstance(match[0]('aa'), AAToken)
+
+
+class VerifyGetTokenTestCase():
+
+    def test_get_token_no_text(self) -> None:
+        lexer: Lexer = Lexer()
+        register_parens(lexer)
+
+        token_match = lexer.tokens.get_token('')
+        assert token_match is None
+
+    def test_get_token_no_text_no_tokens(self) -> None:
+        lexer: Lexer = Lexer()
+        token_match = lexer.tokens.get_token('')
+        assert token_match is None
+
+    def test_get_token_unmatched(self) -> None:
+        lexer: Lexer = Lexer()
+        register_parens(lexer)
+
+        token_match = lexer.tokens.get_token('aaa')
+        assert token_match is None
+
+    def test_get_token_no_tokens(self) -> None:
+        lexer: Lexer = Lexer()
+        register_parens(lexer)
+
+        token_match = lexer.tokens.get_token('aaa')
+        assert token_match is None
+
+    def test_get_token(self) -> None:
+        lexer: Lexer = Lexer()
+        register_parens(lexer)
+
+        match = lexer.tokens.get_token('(((')
+        assert match is not None
+        token_factory_fn, re_match = match
+
+        assert isinstance(token_factory_fn('('), LeftParen)
+        assert re_match is not None
+
+    def test_get_token_picks_first(self) -> None:
+        lexer: Lexer = Lexer()
+        register_parens(lexer)
+
+        token_match = lexer.tokens.get_token('aa(((')
+        assert token_match is None
+
+    def test_get_token_scans_all_possible_tokens(self) -> None:
+        lexer: Lexer = Lexer()
+        register_parens(lexer)
+
+        match = lexer.tokens.get_token(')(')
+        assert match is not None
+        token_factory_fn, re_match = match
+
+        assert isinstance(token_factory_fn(')'), RightParen)
+        assert re_match is not None
+
+    def test_longest_match(self) -> None:
+        lexer: Lexer = Lexer()
+        class AToken(Token):
+            pass
+
+        class AAToken(Token):
+            pass
+
+        lexer.register_token(lambda x: AToken(x), r'a')
+        lexer.register_token(lambda x: AAToken(x), r'aa')
+
+        match = lexer.tokens.get_token('aaa')
+        assert match is not None
+
+        token_factory_fn, re_match = match
+        assert isinstance(token_factory_fn('aa'), AAToken)
+
+
+class VerifyRegisterTokensTestCase():
```

Comment:
@eliotwrobson I think in order for my test runner (`nose2`) to find the tests, the test class must start with `Test` and be subclassed from `unittest.TestCase`. Can you please update this?